projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts

```diff
@@ -1,152 +1,126 @@
   @Input() zoom: number;
   @Input() rotate: number;
   @Input() width: number;
   @Input() height: number;
-  @Input() page: number;
-
-  @ViewChild('container') container: ElementRef;
-
-  selectedAnnotation: SelectionAnnotation = { annotationId: '', editable: false };
+  @ViewChild('boxHighlight') private boxHighlight: BoxHighlightCreateComponent;
+  page: number;
+  selectedAnnotation$: Observable<SelectionAnnotation>;
   drawMode = false;
 
   private subscriptions: Subscription[] = [];
 
   constructor(
+    private store: Store<fromStore.AnnotationSetState>,
     private readonly api: AnnotationApiService,
     private readonly toolbarEvents: ToolbarEventService,
     private readonly viewerEvents: ViewerEventService,
-    private readonly annotationService: AnnotationEventService,
     private readonly commentService: CommentService,
     private readonly boxHighlightService: BoxHighlightCreateService,
     private readonly textHighlightService: TextHighlightCreateService) {}
 
   ngOnInit(): void {
+    this.annotationsPerPage$ = this.store.select(fromStore.getAnnoPerPage)
+      .pipe(tap(annotations => {
+        if (annotations) {
+          this.height = annotations[0].styles.height;
+          this.width = annotations[0].styles.width;
+        }
+    }));
+    this.selectedAnnotation$ = this.store.select(fromStore.getSelectedAnnotation);
+
     this.subscriptions = [
       this.viewerEvents.textHighlight
         .subscribe(highlight => this.createTextHighlight(highlight)),
       this.viewerEvents.boxHighlight
         .subscribe(highlight => this.boxHighlightService.initBoxHighlight(highlight.event)),
-      this.annotationService.getSelectedAnnotation()
-        .subscribe(selectedAnnotation => this.selectedAnnotation = selectedAnnotation),
       this.toolbarEvents.drawModeSubject
         .subscribe(drawMode => this.drawMode = drawMode)
     ];
   }
 
+
   ngOnDestroy(): void {
     this.subscriptions.forEach(subscription => subscription.unsubscribe());
   }
 
-  addToDOM(eventSource: PageEvent['source']) {
-    this.zoom = eventSource.scale;
-    this.rotate = eventSource.rotation;
-    const element = eventSource.div;
-    this.width = this.rotate % 180 === 0 ? element.clientWidth : element.clientHeight;
-    this.height = this.rotate % 180 === 0 ? element.clientHeight : element.clientWidth;
-    element.appendChild(this.container.nativeElement);
-  }
-
   public onAnnotationUpdate(annotation: Annotation) {
-    this.api
-      .postAnnotation(annotation)
-      .subscribe(newAnnotation => {
-        const unsavedComment = this.commentService.hasUnsavedComments(annotation);
-        const index = this.annotationSet.annotations.findIndex(a => a.id === newAnnotation.id);
-
-        this.annotationSet.annotations[index] = unsavedComment ? annotation : newAnnotation;
-        this.selectAnnotation({ annotationId: annotation.id, editable: false });
-      });
+    this.store.dispatch(new fromStore.SaveAnnotation(annotation));
```

Comment:
we might be missing the bit here, where a new annotation is selected when first created

Reply:
that is handled differently now